коммит проекта

alenaa-bva opened the pull request.

alenaa-bva committed:
коммит проекта

AlexKitIt commented on conftest.py:
```diff
@@ -0,0 +1,52 @@
+    yield driver
+    driver.quit()
+
+@pytest.fixture
+def wait_element_to_be_clickable(driver):
+    def wait(locator, timeout=5):
+        return WebDriverWait(driver, timeout).until(expected_conditions.element_to_be_clickable(locator))
+    return wait
+
+@pytest.fixture
+def wait_element_to_be_visible(driver):
+    def wait(locator, timeout=5):
+        return WebDriverWait(driver, timeout).until(expected_conditions.visibility_of_element_located(locator))
+    return wait
```
Нужно исправить: здесь и далее: фикстуры - это предусловия\постусловия тестов а не рандомно переиспользуемые методы. Все это должно быть в тестах, в следующем спринте объяснят куда это стоит положить

AlexKitIt commented on conftest.py:
```diff
@@ -0,0 +1,52 @@
+
+    return register
+
+@pytest.fixture
+def generate_login():
+    def login():
+        base_login = "alenaibragimova1"
+        three_random_digits = random.randint(100, 999)
+        email = f"{base_login}{three_random_digits}@yandex.ru"
+        return email
+    return login
```
Нужно исправить: это необходимо положить в метод в helpers

AlexKitIt commented on tests/test_login_page.py:
```diff
@@ -0,0 +1,147 @@
+            generate_login
+    ):
+        name = "Алена"
+        login = generate_login()
```
Можно лучше: для этих тестов стоит использовать тестового пользователя а не создавать нового каждый раз, так быстрее и удалять его мы не можем

alenaa-bva committed:
- Перенесен генератор логина в файл helpers.py;
- ожидания и метод регистрации убрала в тестовые файлы;
- вместо создания нового пользователя, в тестах используется тестовый с данными из data.py;
- отрефакторила тестовые методы: сократила шаги.

AlexKitIt commented on tests/test_home_page.py:
```diff
@@ -1,7 +1,19 @@
 from locators import LoginPageLocators, HomePageLocators, PersonalAccountPage
 
 
+# вспомогательные функции
+def wait_element_to_be_clickable(driver, locator, timeout=5):
+    return WebDriverWait(driver, timeout).until(expected_conditions.element_to_be_clickable(locator))
+
+def wait_element_to_be_visible(driver, locator, timeout=5):
+    return WebDriverWait(driver, timeout).until(expected_conditions.visibility_of_element_located(locator))
+
+#тесты
```
Нужно исправить: в рамках данного спринта все это пишется в тестах, далее объяснят как это необходимо расположить в проекте
